packages/dashboard/src/app/api/useStoryApi.js

```diff
@@ -153,31 +116,19 @@ const useStoryApi = (dataAdapter, { storyApi }) => {
         trackTiming();
       }
     },
-    [storyApi, dataAdapter, initialFetchListeners]
+    [storyApi, fetchStoriesCallback, initialFetchListeners]
   );
 
   const updateStory = useCallback(
     async (story) => {
       const trackTiming = getTimeTracker('load_update_story');
 
       try {
-        const path = addQueryArgs(`${storyApi}${story.id}/`, {
-          _embed: 'wp:lock,wp:lockuser,author,wp:featuredmedia',
-        });
-
-        const data = {
-          id: story.id,
-          author: story.originalStoryData.author,
-          title: story.title?.raw || story.title,
-        };
-
-        const response = await dataAdapter.post(path, {
-          data,
-        });
+        const response = await updateStoryCallback(story, storyApi);
 
         dispatch({
           type: STORY_ACTION_TYPES.UPDATE_STORY,
-          payload: reshapeStoryObject(response),
+          payload: reshapeStoryObject(response), // @todo Move reshapeStoryObject to wp-dashboard.
```

Comment:
👍

Reply:
I will do it as a separate PR as this PR growing big.